src/main/webapp/app/exam/participate/exam-navigation-sidebar/exam-navigation-sidebar.component.ts

```diff
@@ -0,0 +1,248 @@
+    @Input() examTimeLineView = false;
+    @Input() isTestRun = 0;
+    @Output() onPageChanged = new EventEmitter<{
+        overViewChange: boolean;
+        exercise?: Exercise;
+        forceSave: boolean;
+        submission?: ProgrammingSubmission | SubmissionVersion | FileUploadSubmission;
+    }>();
+
+    static itemsVisiblePerSideDefault = 4;
+    itemsVisiblePerSide = ExamNavigationSidebarComponent.itemsVisiblePerSideDefault;
+
+    subscriptionToLiveExamExerciseUpdates: Subscription;
+
+    // Icons
+    icon: IconProp;
+    overviewIcon: IconProp = faFileLines;
+    faChevronRight = faChevronRight;
+
+    profileSubscription?: Subscription;
+    isProduction = true;
+    isTestServer = false;
+    isCollapsed: boolean = false;
+    exerciseId: string;
+    numberOfSavedExercises: number = 0;
+    isExerciseSaved: boolean[] = new Array(this.exercises.length).fill(false);
+
+    constructor(
+        private profileService: ProfileService,
+        private sidebarEventService: SidebarEventService,
+        private layoutService: LayoutService,
+        private examParticipationService: ExamParticipationService,
+        private examExerciseUpdateService: ExamExerciseUpdateService,
+        private repositoryService: CodeEditorRepositoryService,
+        private conflictService: CodeEditorConflictStateService,
+    ) {}
+
+    ngOnInit(): void {
+        this.profileSubscription = this.profileService.getProfileInfo()?.subscribe((profileInfo) => {
+            this.isProduction = profileInfo?.inProduction;
+            this.isTestServer = profileInfo?.testServer ?? false;
+        });
+
+        if (!this.examTimeLineView) {
+            this.subscriptionToLiveExamExerciseUpdates = this.examExerciseUpdateService.currentExerciseIdForNavigation.subscribe((exerciseIdToNavigateTo) => {
+                // another exercise will only be displayed if the student clicks on the corresponding pop-up notification
+                this.changeExerciseById(exerciseIdToNavigateTo);
+            });
+        }
+
+        this.layoutService.subscribeToLayoutChanges().subscribe(() => {
+            // You will have all matched breakpoints in observerResponse
+            if (this.layoutService.isBreakpointActive(CustomBreakpointNames.extraLarge)) {
+                this.itemsVisiblePerSide = ExamNavigationSidebarComponent.itemsVisiblePerSideDefault;
+            } else if (this.layoutService.isBreakpointActive(CustomBreakpointNames.large)) {
+                this.itemsVisiblePerSide = 3;
+            } else if (this.layoutService.isBreakpointActive(CustomBreakpointNames.medium)) {
+                this.itemsVisiblePerSide = 1;
+            } else {
+                this.itemsVisiblePerSide = 0;
+            }
+        });
+
+        const isInitialSession = this.examSessions && this.examSessions.length > 0 && this.examSessions[0].initialSession;
+        if (isInitialSession || isInitialSession == undefined) {
```

Comment:
What does this accomplish? Would this be enough?
```suggestion
        if (isInitialSession !== false) {
```

Reply:
I didn't write this method, I just copied from `exam-navigation-bar.component.ts` and didn't change anything not to break any functionality. But in my opinion, it might not replace the condition in every edge cases. `if (isInitialSession !== false)` will be true for other falsy values like null, 0, "", and NaN, whereas `if (isInitialSession || isInitialSession == undefined)` will be false for these values.